tests/lib/dependencies.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+# shellcheck source=tests/lib/pkgdb.sh
+. "$TESTSLIB/pkgdb.sh"
+
+export DEPENDENCY_PACKAGES=
+export DISTRO_BUILD_DEPS=
+
+add_pkgs(){
+  DEPENDENCY_PACKAGES="$DEPENDENCY_PACKAGES $@"
+}
+
+get_apt_dependencies_generic(){
+  add_pkgs autoconf
+  add_pkgs automake
+  add_pkgs autotools-dev
+  add_pkgs build-essential
+  add_pkgs indent
+  add_pkgs libapparmor-dev
+  add_pkgs libglib2.0-dev
+  add_pkgs libseccomp-dev
+  add_pkgs libudev-dev
+  add_pkgs pkg-config
+  add_pkgs python3-docutils
+  add_pkgs udev
+}
+
+get_apt_dependencies_classic(){
+  add_pkgs dbus-x11
+  add_pkgs jq
+  add_pkgs man
+  add_pkgs python3-yaml
+  add_pkgs rng-tools
+  add_pkgs upower
+
+  case "$SPREAD_SYSTEM" in
+      ubuntu-14.04-*)
+          add_pkgs cups-pdf
+          add_pkgs linux-image-extra-$(uname -r)
+          add_pkgs pollinate
+          ;;
+      ubuntu-16.04-32)
+          add_pkgs linux-image-extra-$(uname -r)
+          add_pkgs pollinate
+          add_pkgs printer-driver-cups-pdf
+          ;;
+      ubuntu-16.04-64)
+          add_pkgs gccgo-6
+          add_pkgs kpartx
+          add_pkgs libvirt-bin
+          add_pkgs linux-image-extra-$(uname -r)
+          add_pkgs pollinate
+          add_pkgs printer-driver-cups-pdf
+          add_pkgs qemu
+          add_pkgs x11-utils
+          add_pkgs xvfb
+          ;;
+      ubuntu-*)
+          add_pkgs linux-image-extra-$(uname -r)
+          add_pkgs pollinate
+          add_pkgs printer-driver-cups-pdf
+          ;;
+      debian-*)
+          add_pkgs printer-driver-cups-pdf
+          ;;
+  esac 
+}
+
+get_apt_dependencies_core(){
+  add_pkgs linux-image-extra-$(uname -r)
+  add_pkgs pollinate
+}
+
+get_dependency_fedora_packages(){
+  echo "Fedora dependencies not ready yet"
+}
+
+get_dependency_opensuse_packages(){
+  echo "Opensuse dependencies not ready yet"
+}
+
+get_test_dependencies(){
```

Comment:
Is there an advantage in splitting down test and build?  Won't we end up with both of these installed anyway?